functions/inputs/from.go

```diff
@@ -128,13 +113,48 @@ type FromProcedureSpec struct {
 
 	GroupingSet bool
 	OrderByTime bool
-	GroupMode   GroupMode
+	GroupMode   functions.GroupMode
 	GroupKeys   []string
 
 	AggregateSet    bool
 	AggregateMethod string
 }
 
+// FromRangeTransformationRule pushes a `range` into a `from`
+type FromRangeTransformationRule struct{}
+
+// Name returns the name of the rule
```

Comment:
You can register this rule in the `init` method above so the planner picks it up by default.  Then we can begin testing the new planner against an actual running database.